pynamodb/expressions/util.py

```diff
@@ -0,0 +1,29 @@
+import re
+
+PATH_SEGMENT_REGEX = re.compile(r'([^\[\]]+)((?:\[\d+\])*)$')
+
+
+def substitute_names(expression, placeholders):
+    """
+    Replaces names in the given expression with placeholders.
+    Stores the placeholders in the given dictionary.
+    """
+    path_segments = expression.split('.')
```

Comment:
Would this break if we pass in an attribute name that has a '.' in it? Or was this already not working in Pynamodb before